File: tchannel/messages/call_request.py
from __future__ import absolute_import

from .base import BaseMessage
from .types import Types


class CallRequestMessage(BaseMessage):
    """Initiate an RPC call."""
    message_type = Types.CALL_REQ

    __slots__ = (
        'flags',
        'ttl',

        # Zipkin-style tracing data
        'span_id',
        'parent_id',
        'trace_id',

        'traceflags',
        'service',
        'headers',

        'arg1',
        'arg2',
        'arg3',
    )

File: java/tchannel-core/src/main/java/com/uber/tchannel/messages/CallRequest.java
package com.uber.tchannel.messages;

import com.uber.tchannel.tracing.Trace;

import java.util.Map;

public class CallRequest extends AbstractCallMessage {

    private final long ttl;
    private final Trace tracing;
    private final String service;
    private final Map<String, String> headers;

    public CallRequest(long id, byte flags, long ttl, Trace tracing, String service,
                       Map<String, String> headers, byte checksumType, int checksum,
                       byte[] arg1, byte[] arg2, byte[] arg3) {
        super(id, MessageType.CallRequest, flags, checksumType, checksum, arg1, arg2, arg3);
        this.ttl = ttl;
        this.service = service;
        this.tracing = tracing;
        this.headers = headers;
    }
}
